src/main/java/org/mskcc/cbio/portal/dao/DaoSampleList.java

```diff
@@ -248,30 +226,46 @@ private int addSampleListList(SampleList sampleList, Connection con) throws DaoE
         try {
             StringBuilder sql = new StringBuilder("INSERT INTO sample_list_list (`LIST_ID`, `SAMPLE_ID`) VALUES ");
             // NOTE - as of 12/12/14, patient lists contain sample ids
-            for (String sampleId : sampleList.getSampleList()) {
-                Sample sample = DaoSample.getSampleByCancerStudyAndSampleId(sampleList.getCancerStudyId(), sampleId);
+            for (String sampleId : sampleList) {
+                Sample sample = DaoSample.getSampleByCancerStudyAndSampleId(cancerStudyId, sampleId);
                 if (sample == null) {
-                    System.out.println("null sample: " + sampleId + ":" + sampleList.getStableId());
+                    System.out.println("null sample: " + sampleId);
                     ++skippedPatients;
                     continue;
                 }
                 sql.append("('").append(sampleListId).append("','").append(sample.getInternalId()).append("'),");
             }
-            if (skippedPatients == sampleList.getSampleList().size()) {
+            if (skippedPatients == sampleList.size()) {
                 return 0;
             }
             sql.deleteCharAt(sql.length()-1);
             pstmt = con.prepareStatement(sql.toString());
             return pstmt.executeUpdate();
-        } catch (NullPointerException e) {
-            throw new DaoException(e);
         } catch (SQLException e) {
             throw new DaoException(e);
         } finally {
             JdbcUtil.closeAll(rs);
         }
     }
 
+    public void updateSampleListList(SampleList sampleList) throws DaoException {
+        Connection con = null;
+        PreparedStatement pstmt = null;
+        try {
+            con = JdbcUtil.getDbConnection(DaoSampleList.class);
+            pstmt = con.prepareStatement(DELETE_SAMPLE_LIST_LIST);
+            pstmt.setInt(1, sampleList.getSampleListId());
+            pstmt.executeUpdate();
+
+            addSampleListList(sampleList.getCancerStudyId(), sampleList.getSampleListId(), sampleList.getSampleList(), con);
```

Comment:
This is a complete delete / replace for the records listing the samples in a sample list.

Reply:
Do you anticipate any problems by doing an update with delete+insert?

Reply:
No .. no problems. My comment was only documenting my understanding during my read through.